x-pack/plugins/security_solution/server/lib/detection_engine/routes/fleet/get_installed_integrations/get_installed_integrations_route.ts

```diff
@@ -39,17 +46,26 @@ export const getInstalledIntegrationsRoute = (router: SecuritySolutionPluginRout
           set.addPackagePolicy(policy);
         });
 
-        const registryPackages = await Promise.all(
-          set.getPackages().map((packageInfo) => {
-            return fleet.packages.getRegistryPackage(
+        const registryPackages = await initPromisePool({
+          concurrency: MAX_CONCURRENT_REQUESTS_TO_PACKAGE_REGISTRY,
+          items: set.getPackages(),
+          executor: async (packageInfo) => {
+            const registryPackage = await fleet.packages.getRegistryPackage(
               packageInfo.package_name,
               packageInfo.package_version
             );
-          })
-        );
+            return registryPackage;
+          },
+        });
+
+        registryPackages.errors.forEach(({ error, item }) => {
+          const logMessage = `Error fetching package info from registry for ${item.package_name}@${item.package_version}`;
+          const logReason = error instanceof Error ? error.message : String(error);
+          logger.error(`${logMessage}. ${logReason}`);
```

Comment:
While nice we're logging these now, I'm curious how actionable this will be for users, and potential overall noisiness of these logs as well. Perhaps `debug` may be a better log level for now?

Reply:
I think it depends on the error. Something like `fleet_server@1.2.0 not found` as in the GH issue probably wouldn't be actionable for users. But if a user has a custom package registry, and it's not available, it would likely end up with an exception here and let the user know that they need to fix their environment.

It could be noisy, I agree. I think we have a similar situation with `IRuleExecutionLogForRoutes` and in its implementation. There, we don't know for sure what exceptions we might catch (it could be a network error, a bug in the underlying code of SO Client or Event Log Reader, or a business error), but we log them via `logger.error`. Would you rather log them on the `debug` level?

I don't have a strong opinion on this, both options are ok to me.

Reply:
Improved console logs a bit in https://github.com/elastic/kibana/pull/134732/commits/d6a6a461dd6ac7e8f7abc2132cc7d20974b301a8

Errors should be less noisy but still visible to users by default. Added `Unable to retrieve installed integrations` for more context.

```
[2022-06-20T21:03:44.551+02:00][ERROR][plugins.securitySolution] Unable to retrieve installed integrations. Error fetching packages from registry: fleet_server@1.2.0, apm@8.4.0.
[2022-06-20T21:03:44.551+02:00][DEBUG][plugins.securitySolution] Error fetching package info from registry for fleet_server@1.2.0. Boom!
[2022-06-20T21:03:44.551+02:00][DEBUG][plugins.securitySolution] Error fetching package info from registry for apm@8.4.0. Boom!
```